modules/analysis/rosbag-image-pipeline/deployspec.yaml

```diff
@@ -22,7 +22,8 @@ deploy:
       # Copy DAG files to S3
       - aws s3 cp --recursive $DAG_DIR/ s3://$ADDF_PARAMETER_DAG_BUCKET_NAME/$ADDF_PARAMETER_DAG_PATH/$DAG_DIR/
       # - Building the Docker image(s)...
-      - aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin 763104351884.dkr.ecr.$AWS_DEFAULT_REGION.amazonaws.com
+      # - The region here matches the region used in the Dockerfile for pulling AWS Deep Learning Containers
+      - aws ecr get-login-password --region eu-central-1 | docker login --username AWS --password-stdin 763104351884.dkr.ecr.eu-central-1.amazonaws.com
```

Comment:
You seem to have hardcoded the `region` and `account`. Will this work from a cross-region?

Reply:
Yes exactly, for the sagemaker Deep Learning containers,  you have to login to this speciific AWS account to pull them. And the region here has to match the region in the top of the Dockerfile. Since we can't put variables in the Dockerfile, I hardcode it as eu-central-1, which I tested in eu-west-1 to make sure this works. And the account is also stable. More info on this here: https://github.com/aws/deep-learning-containers/blob/master/available_images.md

Reply:
Instead of capturing the region details inside Dockerfile, what you can do is, declare the `BASE_IMAGE` as a Docker Arg:

```
ARG BASE_IMAGE
```

Then from your deployspec, pass the arg value during docker build time as:

```
docker build --build-arg BASE_IMAGE=<<BASE_IMAGE_VALUE>>
```